android/app/src/main/java/com/zulipmobile/notifications/NotificationsModule.java

```diff
@@ -40,11 +40,12 @@ public void getToken(Promise promise) {
     }
 
     @ReactMethod
-    public void getInitialNotification(Promise promise) {
+    public void readInitialNotification(Promise promise) {
         if (null == initialNotification) {
             promise.resolve(null);
         } else {
             promise.resolve(Arguments.fromBundle(initialNotification));
         }
+        initialNotification = null;
```

Comment:
```suggestion
    public void readInitialNotification(Promise promise) {
        if (null == initialNotification) {
            promise.resolve(null);
        } else {
            promise.resolve(Arguments.fromBundle(initialNotification));
            initialNotification = null;
        }
```

This has the same behavior but I think is a bit cleaner -- it more closely ties the act of clearing `initialNotification` with the act of passing its contents on elsewhere. That makes it more like just a single atomic act of consuming it, which is how we want to think of this.